intellij/src/main/java/toolWindow/MyToolWindow.java

```diff
@@ -77,18 +90,76 @@ public JComponent getInput() {
-                String clientClassNamespace = postClientNameField.getText();
-                KiotaGenerationLanguage selectedLanguage = (KiotaGenerationLanguage) languageComboBox.getSelectedItem();
-                generateClientHandler.generateclient(descriptionPath, output, selectedLanguage, include, exclude, clientClass, clientClassNamespace);
+                generateButton.setVisible(false);
+                progressBar.setVisible(true);
+                progressBarPanel.removeAll(); // Clear any previous components
+                progressBarPanel.add(progressBar); // Add the progress bar
+
+                // perform the generation in a separate thread
+                SwingWorker<Void, Void> worker = new SwingWorker<Void, Void>() {
+                    @Override
+                    protected Void doInBackground() throws Exception {
+                        String descriptionPath = yamlFilePathField.getText();
+                        String output = outputpath.getText();
+                        String clientClass = clientClassField.getText();
+                        String clientClassNamespace = postClientNameField.getText();
+                        KiotaGenerationLanguage selectedLanguage = (KiotaGenerationLanguage) languageComboBox.getSelectedItem();
+
+                        generateClientHandler.generateclient(descriptionPath, output, selectedLanguage, include, exclude, clientClass, clientClassNamespace);
+
+                        return null;
+                    }
+                    @Override
+                    protected void done() {
+                        // Generation is complete, re-enable the button
+                        progressBarPanel.setVisible(true);
+                        Object response = generateClientHandler.getResponse();
+                        String responseString = response.toString();
+
+                        Project[] openProjects = ProjectManager.getInstance().getOpenProjects();
+                        if (openProjects.length > 0) {
+                            Project currentProject = openProjects[0]; // Get the first open project
+
+                            if (responseString.contains("error generating the client")) {
+                                showStickyNotification(currentProject, "Error generating the client. " + responseString, NotificationType.ERROR);
+                                progressBar.setVisible(false);
+                                progressBar.setEnabled(false);
+                                generateButton.setVisible(true);
+                            } else if (responseString.contains("completed")) {
+                                showStickyNotification(currentProject, "Generation completed successfully! ", NotificationType.INFORMATION);
+                                progressBar.setVisible(false);
+                                generateButton.setVisible(true);
+                            }
```

Comment:
What if `responseString.contains("error generating the client")` is not true, and `responseString.contains("completed")` is not true? 

We should make sure that we can handle the unexpected case that neither of those scenarios are true. Unexpected conditions happen so we should consider how we should handle them.

Reply:
tracking with #3153

Reply:
tracking with #3169